src/chainrules.jl

```diff
@@ -193,3 +193,63 @@ ChainRulesCore.@scalar_rule(
 ChainRulesCore.@scalar_rule(expinti(x), exp(x) / x)
 ChainRulesCore.@scalar_rule(sinint(x), sinc(invπ * x))
 ChainRulesCore.@scalar_rule(cosint(x), cos(x) / x)
+
+# non-holomorphic functions
+function ChainRulesCore.frule((_, _, _), ::typeof(besselix), ν::Number, x::Number)
+    Ω = besselix(ν, x)
+    ΔΩ = ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO)
+    return Ω, ΔΩ
```

Comment:
My earlier comment might have gotten lost, but since `ZeroTangent() * ::NotImplemented` is a `ZeroTangent`, you can implement the `frule` in such a way that if the AD provides `ZeroTangent()` for `Δν`, then the `frule` does not return a `NotImplemented`:
```suggestion
function ChainRulesCore.frule((_, Δν, Δx), ::typeof(besselix), ν::Number, x::Number)
    Ω = besselix(ν, x)
    ∂Ω_∂ν = ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO)
    a = (besselix(ν - 1, x) + besselix(ν + 1, x)) / 2
    ∂Ω = muladd(a, Δx, muladd(-sign(real(x)) * real(Δx), Ω, ∂Ω_∂ν * Δν)
    return Ω, ∂Ω
```

Reply:
Oh yes, I completely forgot that `ZeroTangent() * ::NotImplemented = ZeroTangent()`. I'll fix the forward rules!

Reply:
I noticed that currently it is not possible to test such definitions: https://github.com/JuliaDiff/ChainRulesCore.jl/pull/477